homeassistant/components/system_bridge/coordinator.py

```diff
@@ -49,97 +82,127 @@ def update_listeners(self) -> None:
         for update_callback in self._listeners:
             update_callback()
 
-    async def async_handle_event(self, event: Event):
-        """Handle System Bridge events from the WebSocket."""
+    async def async_handle_message(self, message: Generic) -> None:
+        """Handle messages from the WebSocket."""
         # No need to update anything, as everything is updated in the caller
-        self.logger.debug(
-            "New event from %s (%s): %s", self.title, self.host, event.name
-        )
-        self.async_set_updated_data(self.bridge)
+        self.logger.debug("New message from %s: %s", self.title, message[EVENT_TYPE])
+        if message[EVENT_TYPE] == TYPE_DATA_UPDATE:
+            self.logger.debug("Set new data for: %s", message[EVENT_MODULE])
+            if message[EVENT_MODULE] == "battery":
+                self.systembridge_data.battery = Battery(**message[EVENT_DATA])
+            elif message[EVENT_MODULE] == "cpu":
```

Comment:
We'd benefit from using an event handler approach to solve this code repetition.

```py
setattr(self.systembridge_data, message.event_module, message.event_model)
```

If we need different handlers we can do it like this:
https://github.com/home-assistant/core/blob/77d39f9c4f4b10bfab25eb850b5e26542d8cbcdc/homeassistant/components/overkiz/coordinator.py#L101-L102

https://github.com/home-assistant/core/blob/77d39f9c4f4b10bfab25eb850b5e26542d8cbcdc/homeassistant/components/overkiz/coordinator.py#L127-L133